src/Functions/JSONPath/Generators/GeneratorJSONPath.h

```diff
@@ -0,0 +1,103 @@
+#include <Functions/JSONPath/Generators/IGenerator.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathMemberAccess.h>
+#include <Functions/JSONPath/Generators/VisitorJSONPathRange.h>
+#include <Functions/JSONPath/Generators/VisitorStatus.h>
+
+#include <Functions/JSONPath/ASTs/ASTJSONPath.h>
+
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int LOGICAL_ERROR;
+}
+
+template <typename JSONParser>
+class GeneratorJSONPath : public IGenerator<JSONParser>
+{
+public:
+    GeneratorJSONPath(ASTPtr query_ptr_)
+    {
+        query_ptr = query_ptr_;
+        const auto * path = query_ptr->as<ASTJSONPath>();
+        if (!path) {
+            throw Exception("Invalid path", ErrorCodes::LOGICAL_ERROR);
+        }
+        const auto * query = path->jsonpath_query;
+
+        for (auto child_ast : query->children)
+        {
+            if (child_ast->getID() == "ASTJSONPathMemberAccess")
```

Comment:
But then we would have to define in all of ASTs both getID (since is it purely virtual) and this over getSomething, which returns enum. In my opinion it would be even more awkward that what we have done

Reply:
See solution below with `typeid_cast`.